weird use case: Crash when connecting wallet to Dapp

Steps to reproduce -

1. Generate a new account A
2. Go to browser and connect dapps to account A
3. Go Profile and check if dapps only has permission to A
4. Delete the account A form wallet settings
5. Go to Wallet create account B
6. In Browser try to connect account B to dapps.

the app crashes 

ml: true
DBG 2022-03-24 10:35:40.101+01:00 NewBE_callPrivateRPC                       topics="rpc" tid=291026 file=core.nim:20 rpc_method=permissions_addDappPermissions
INF 2022-03-24 10:35:40.104+01:00 shutting down...                           topics="status-app" tid=291026 file=nim_status_client.nim:119
Object 0x55c1ec5c7650 destroyed while one of its QML signal handlers is in progress.
Most likely the object was deleted synchronously (use QObject::deleteLater() instead), or the application is running a nested event loop.
This behavior is NOT supported!
qrc:/app/AppLayouts/Browser/popups/BrowserConnectionModal.qml:180: function() { [native code] }
Traceback (most recent call last)
/home/khushboo/status/nim-status-client/src/nim_status_client.nim(149) nim_status_client
/home/khushboo/status/nim-status-client/src/nim_status_client.nim(128) mainProc
/home/khushboo/status/nim-status-client/src/app/boot/app_controller.nim(228) delete
/home/khushboo/status/nim-status-client/src/app/global/global_singleton.nim(67) delete
/home/khushboo/status/nim-status-client/vendor/nimqml/src/nimqml/private/qqmlapplicationengine.nim(40) delete
SIGABRT: Abnormal termination.
make: *** [Makefile:577: run-linux] Error 1

--- a/src/app/modules/main/browser_section/dapps/module.nim
+++ b/src/app/modules/main/browser_section/dapps/module.nim
@@ -46,9 +46,13 @@ method fetchDapps*(self: Module) =
     var found = false
     for item in items:
       if item.name == dapp.name:
-        let account = self.controller.getAccountForAddress(dapp.address)
-        item.addAccount(account)
         found = true
+
+        let account = self.controller.getAccountForAddress(dapp.address)
+        if account.isNil:
+          break
+
+        item.addAccount(account)
         break
       
     if not found:

--- a/src/app_service/service/wallet_account/service.nim
+++ b/src/app_service/service/wallet_account/service.nim
@@ -213,6 +213,9 @@ proc init*(self: Service) =
     return
 
 proc getAccountByAddress*(self: Service, address: string): WalletAccountDto =
+  if not self.accounts.hasKey(address):
+    return
+
   return self.accounts[address]
 
 proc getWalletAccounts*(self: Service): seq[WalletAccountDto] =

